### PR TITLE
Sak 29914 Making calendar export period configurable by adding config params

### PR DESCRIPTION
**Pull request by surya-janani**

### CODE REVIEW

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 1]
```diff
@@ -6375,15 +6375,24 @@ protected int generateICal(net.fortuna.ical4j.model.Calendar ical, List<String>
 	}
 	
 	/* Given a current date via the calendarUtil paramter, returns a TimeRange for the year,
-	 * 6 months either side of the current date. (calculate milleseconds in 6 months)
+	  *fromMonthsInput number of months from past to be included
++	  *toMonthsInput number of months in future  to be included.
 	 */
    private static long SIX_MONTHS = (long)1000 * (long)60 * (long)60 * (long)24 * (long)183;
```
Comment (ottenhoff) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java]: is this SIX_MONTHS constant still being used?

**Comment by surya-janani** [comment]
No. I've deleted that line now

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 2]
```diff
@@ -6375,15 +6375,22 @@ protected int generateICal(net.fortuna.ical4j.model.Calendar ical, List<String>
...
 	public TimeRange getICalTimeRange()
 	{
-		Time now = m_timeService.newTime();
-		Time startTime = m_timeService.newTime( now.getTime() - SIX_MONTHS );
-		Time endTime = m_timeService.newTime( now.getTime() + SIX_MONTHS );
+		int toMonthsInput = m_serverConfigurationService.getInt("calendar.export.next.months",6);
+		int fromMonthsInput = m_serverConfigurationService.getInt("calendar.export.previous.months",6);
+
+		java.util.Calendar calTo = java.util.Calendar.getInstance();
+				calTo.add(java.util.Calendar.MONTH, toMonthsInput);
```
Comment (ottenhoff) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java]: Can you fix the spacing here?

**Comment by surya-janani** [comment]
Corrected indendation and rebased

**Pull request merged**